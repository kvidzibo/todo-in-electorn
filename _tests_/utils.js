import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

export const render = (comp, mode = 3) => {
  switch (mode) {
    case 1:
      return shallow(comp)
    case 2:
      return renderer.create(comp).toJSON()
    case 3:
      return {
        component: shallow(comp),
        tree: renderer.create(comp).toJSON()
      }
    case 4:
      return mount(comp)
    case 5:
      return {
        component: mount(comp),
        tree: renderer.create(comp).toJSON()
      }
  }
}
