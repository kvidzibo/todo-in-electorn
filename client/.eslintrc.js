module.exports = {
    "extends": "standard",
    "globals": {$: true},
    "plugins": [
      "react"
    ],
    "parser": "babel-eslint",
    "rules": {
      "react/jsx-uses-vars": 1,
      "react/jsx-uses-react": 1
    }
};
