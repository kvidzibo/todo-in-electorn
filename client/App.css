.ml5 {
    margin-left: 0.5rem;
}
