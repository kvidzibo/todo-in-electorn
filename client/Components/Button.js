import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

class Button extends PureComponent {
  render () {
    const { children, onClick, control, icon } = this.props
    const buttonComp = (
      <button onClick={onClick} className="button">
        {icon ? <span className="icon"><i className={`fa fa-${icon}`}></i></span> : children}
      </button>
    )
    return control ? (
      <div className="control">
        {buttonComp}
      </div>
    ) : (buttonComp)
  }
}

Button.propTypes = {
  icon: PropTypes.string,
  children: PropTypes.node,
  onClick: PropTypes.func,
  control: PropTypes.bool
}

Button.defaultProps = {
  icon: '',
  control: false
}

export default Button
