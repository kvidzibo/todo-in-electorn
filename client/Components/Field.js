import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class Field extends PureComponent {
  render () {
    const { label, children, error, grouped, uncontrolled } = this.props
    const className = classNames({
      'field': true,
      'is-grouped': grouped
    })
    return (
      <div className={className}>
        {!!label && (<label className="label">{label}</label>)}
        {uncontrolled ? children : <div className="control">{children}</div>}
        {!!error && <p className="help is-danger">{error}</p>}
      </div>
    )
  }
}

Field.propTypes = {
  label: PropTypes.string,
  children: PropTypes.node.isRequired,
  error: PropTypes.string,
  grouped: PropTypes.bool,
  uncontrolled: PropTypes.bool
}

Field.defaultProps = {
  label: '',
  error: '',
  grouped: false,
  uncontrolled: false
}

export default Field
