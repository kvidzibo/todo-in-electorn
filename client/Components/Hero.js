import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class Hero extends PureComponent {
  render () {
    const { title, subtitle, primary } = this.props
    const className = classNames({
      'hero': true,
      'is-primary': primary
    })

    return (
      <section className={className}>
        <div className="hero-body">
          <div className="container">
            {!!title && (
              <h1 className="title">{title}</h1>
            )}
            {!!subtitle && (
              <h2 className="subtitle">{subtitle}</h2>
            )}
          </div>
        </div>
      </section>
    )
  }
}

Hero.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  primary: PropTypes.bool
}

Hero.defaultProps = {
  title: '',
  subtitle: '',
  primary: false
}

export default Hero
