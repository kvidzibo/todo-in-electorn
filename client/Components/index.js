export {default as Hero} from './Hero'
export {default as Section} from './Section'
export {default as Field} from './Field'
export {default as Button} from './Button'
