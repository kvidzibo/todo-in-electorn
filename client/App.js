import React, { PureComponent } from 'react'

import 'bulma/css/bulma.css'
import 'font-awesome/css/font-awesome.min.css'
import './App.css'

import { Hero, Section } from './Components'
import { withRefetch } from './HOC'
import { List as TodoList, Module as TodoM } from './Todo'

class App extends PureComponent {
  render () {
    const { data } = this.props

    return (
      <div className="app">
        <Hero
          title="Todo in electron"
          subtitle="The best todo app ever!"
          primary
        />
        <Section title="List" subtitle={`todo: ${data[1]} done: ${data[0]}`}>
          <TodoList />
        </Section>
      </div>
    )
  }
}

export default withRefetch(App, TodoM, () => Promise.all([
  TodoM.count({status: 'done'}),
  TodoM.count({status: 'todo'})
]))
