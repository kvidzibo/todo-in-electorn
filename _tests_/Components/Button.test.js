import React from 'react'
import { render } from '../utils'

import { Button } from '../../client/Components'

test('displays childern', () => {
  const { component, tree } = render(<Button>test</Button>)
  expect(component.text()).toEqual('test')
  expect(tree).toMatchSnapshot()
})

test('displays icon', () => {
  const { component, tree } = render(<Button icon="test"/>)
  expect(component.find('i').length).toEqual(1)
  expect(tree).toMatchSnapshot()
})

test('adds control', () => {
  const { component, tree } = render(<Button control icon="test"/>)
  expect(component.find('div.control').length).toEqual(1)
  expect(tree).toMatchSnapshot()
})
