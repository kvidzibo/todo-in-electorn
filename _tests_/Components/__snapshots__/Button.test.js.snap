// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds control 1`] = `
<div
  className="control"
>
  <button
    className="button"
    onClick={undefined}
  >
    <span
      className="icon"
    >
      <i
        className="fa fa-test"
      />
    </span>
  </button>
</div>
`;

exports[`displays childern 1`] = `
<button
  className="button"
  onClick={undefined}
>
  test
</button>
`;

exports[`displays icon 1`] = `
<button
  className="button"
  onClick={undefined}
>
  <span
    className="icon"
  >
    <i
      className="fa fa-test"
    />
  </span>
</button>
`;
